src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+
+        if( thisRuntime.runtimeInfo.stream().mapToInt(entry -> entry._2()._1().ordinal()).sum()!=0 ){
+            return null;
+        }else{
+            return thisRuntime.fileNameToReturn;
+        }
+    }
+
+    @VisibleForTesting
+    static SGAUnitRuntimeInfoStruct runSGAPreprocess(final Path sgaPath,
+                                                     final File inputFASTQFile,
+                                                     final File outputDirectory,
+                                                     final SGAModule indexer,
+                                                     final ArrayList<String> indexerArgs){
+
+        final String prefix = FilenameUtils.getBaseName(inputFASTQFile.getName());//extractBaseNameWithoutExtension(inputFASTQFile);
+        final String preprocessedFASTAFileName = prefix+".pp.fa";
+
+        final SGAModule preprocess = new SGAModule("preprocess");
+        final ArrayList<String> ppArgs = new ArrayList<>();
+        ppArgs.add("--pe-mode");    ppArgs.add("2");
+        ppArgs.add("--pe-orphans"); ppArgs.add(prefix+".pp.orphan.fa");
+        ppArgs.add("--out");        ppArgs.add(preprocessedFASTAFileName);
+        ppArgs.add(inputFASTQFile.getName());
+
+        final Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String> ppInfo = preprocess.run(sgaPath, outputDirectory, ppArgs);
+
+        indexerArgs.set(indexerArgs.size()-1, preprocessedFASTAFileName);
+        final Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String> indexerInfo = indexer.run(sgaPath, outputDirectory, indexerArgs);
+
+        final SGAUnitRuntimeInfoStruct ppUnitInfo = new SGAUnitRuntimeInfoStruct("preprocess", preprocessedFASTAFileName, ppInfo._1(), ppInfo._2(), ppInfo._3());
+        ppUnitInfo.add("index", indexerInfo._1(), indexerInfo._2(), indexerInfo._3());
+
+        return ppUnitInfo;
+    }
+
+    @VisibleForTesting
+    static SGAUnitRuntimeInfoStruct runSGACorrect(final Path sgaPath,
+                                                  final File inputFASTAFile,
+                                                  final File outputDirectory,
+                                                  final int threads,
+                                                  final SGAModule indexer,
+                                                  final ArrayList<String> indexerArgs){
+        return runSimpleModuleFollowedByIndexing(sgaPath, "correct", ".ec.fa", inputFASTAFile, outputDirectory, threads, indexer, indexerArgs);
+    }
+
+    @VisibleForTesting
+    static SGAUnitRuntimeInfoStruct runSGAFilter(final Path sgaPath,
+                                                 final File inputFASTAFile,
+                                                 final File outputDirectory,
+                                                 final int threads){
+        final String prefix = FilenameUtils.getBaseName(inputFASTAFile.getName());//extractBaseNameWithoutExtension(inputFASTAFile);
```

Comment:
`sga filter` actually automatically builds index, so it's not necessary to manually call `sga index` on the filter-passing sequences.